client/descriptions.go

```diff
@@ -30,6 +32,13 @@ type CollectionDescription struct {
 	// The ID of the schema version that this collection is at.
 	SchemaVersionID string
 
+	// BaseQuery contains the base query of this view, if this is collection is a view.
```

Comment:
Cheers :)

- [ ] "is this collection is" typo

> Should it be "Actor-defined"?

I don't think so, why?

Reply:
> I don't think so, why do you think so?

Otherwise it the sentence sound unnatural (at least to my non-native ears). Why then you used past tense "Actor defined"?

Reply:
> Why then you used past tense "Actor defined"?

They are aggregates that have been (past tense), by an actor.  Both with and without hyphen made sense to me, and I think I over-use special characters when writing, and so I try and avoid them.

Grammarly suggests using hyphens like you suggested here: https://www.grammarly.com/blog/hyphen/ though.  I'm not sure it is a hard grammar rule in English though, Collins (respected author of dictionaries) keeps their usage simple: https://grammar.collinsdictionary.com/easy-learning/when-do-you-use-a-hyphen-in-english.

Reply:
good arguments, thanks. I'm not trying to say that it's wrong or not. Just giving you my perception.
My judgement based mostly on my own understanding. If I had to read it few time to make sure I understand, then some others will do as well.